src/connectors/impls/file.rs

```diff
@@ -250,15 +260,19 @@ struct FileSink {
 }
 
 impl FileSink {
-    async fn new(config: &Config, files_to_read: Sender<FileDescriptor>) -> Result<Self> {
+    async fn new(
+        config: &Config,
+        initial_command: &Option<Command>,
```

Comment:
I think it would be nicer to pull this out into the wrapping connector abstraction itself, that way we don't need to implement it for each source/sink repeatedly. Instead the wrapper could send the initial commands as messages to control upon startup making the whole initial commands logic opaque to the connector implementer.